bundles/org.openhab.binding.ntp/src/main/java/org/openhab/binding/ntp/internal/handler/NtpHandler.java

```diff
@@ -54,217 +54,183 @@
+        this.timeZoneId = defaultTimeZoneId;
     }
 
     @Override
     public void handleCommand(ChannelUID channelUID, Command command) {
-        // No specific commands tied to this, but we will trigger an update
-        this.refreshNtpCount = 0;
-        refreshTimeDate();
+        if (command == RefreshType.REFRESH) {
+            logger.debug("Refreshing channel '{}' for '{}'.", channelUID.getId(), getThing().getUID());
+            refreshTimeDate();
+        }
     }
 
     @Override
     public void initialize() {
-        try {
-            logger.debug("Initializing NTP handler for '{}'.", getThing().getUID());
+        logger.debug("Initializing NTP handler for '{}'.", getThing().getUID());
 
-            Configuration config = getThing().getConfiguration();
-            hostname = config.get(PROPERTY_NTP_SERVER_HOST).toString();
-            port = (BigDecimal) config.get(PROPERTY_NTP_SERVER_PORT);
-            refreshInterval = (BigDecimal) config.get(PROPERTY_REFRESH_INTERVAL);
-            refreshNtp = (BigDecimal) config.get(PROPERTY_REFRESH_NTP);
-            refreshNtpCount = 0;
+        NtpThingConfiguration config = getConfigAs(NtpThingConfiguration.class);
+        configuration = config;
 
-            try {
-                Object timeZoneConfigValue = config.get(PROPERTY_TIMEZONE);
-                if (timeZoneConfigValue != null) {
-                    timeZone = TimeZone.getTimeZone(timeZoneConfigValue.toString());
-                } else {
-                    timeZone = TimeZone.getDefault();
-                    logger.debug("{} using default TZ '{}', because configuration property '{}' is null.",
-                            getThing().getUID(), timeZone, PROPERTY_TIMEZONE);
-                }
-            } catch (Exception e) {
-                timeZone = TimeZone.getDefault();
-                logger.debug("{} using default TZ '{}' due to an occurred exception: ", getThing().getUID(), timeZone,
-                        e);
-            }
+        refreshNtpCount = 0;
 
+        String timeZoneConfigValue = config.getTimeZone();
+        if (timeZoneConfigValue != null) {
+            logger.debug("{} with timezone '{}' set in configuration setting '{}'", getThing().getUID(),
+                    timeZoneConfigValue, NtpThingConfiguration.TIMEZONE);
             try {
-                Object localeStringConfigValue = config.get(PROPERTY_LOCALE);
-                if (localeStringConfigValue != null) {
-                    locale = new Locale(localeStringConfigValue.toString());
-                } else {
-                    locale = localeProvider.getLocale();
-                    logger.debug("{} using default locale '{}', because configuration property '{}' is null.",
-                            getThing().getUID(), locale, PROPERTY_LOCALE);
-                }
+                timeZoneId = TimeZone.getTimeZone(timeZoneConfigValue).toZoneId();
             } catch (Exception e) {
```

Comment:
Is it necessary to catch `Exception` here?

Reply:
You're right, it is not necessary because it returns GMT if the string is invalid.
Fixed.